src/vm/jitinterfacegen.cpp

```diff
@@ -224,10 +224,10 @@ void InitJITHelpers1()
         if (GCHeap::UseAllocationContexts())
         {
 #ifdef FEATURE_IMPLICIT_TLS
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST_ALIGN8, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_VC, JIT_NewArr1VC_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_OBJ, JIT_NewArr1OBJ_MP_FastPortable);
+            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, reinterpret_cast<void*>(JIT_NewS_MP_FastPortable));
```

Comment:
I don't think that's a good guideline. Unless I misunderstand it, in its extreme form, it would mean bloating code with many checks all over the place and in case a function changes its contract (say it is a function that was accepting numbers from 0..10 and now accepts numbers 0..100), fixing the checks at all the callers. It seems to me that the opposite - checking the parameter validity inside of a function - is a better practice. Then the function developer is responsible for ensuring that only valid input is accepted and is able to update that.
However, in the current case, it doesn't seem like an argument correctness issue. Here the macro is supposed to set a helper function address, no matter what the signature is. So I agree with @jkotas that casting it inside of the macro is a better way.